runelite-mixins/src/main/java/net/runelite/mixins/RSClientMixin.java

```diff
@@ -1245,8 +1248,13 @@ public static void onAddChatMessage(int type, String name, String message, Strin
 			logger.debug("Chat message type {}: {}", ChatMessageType.of(type), message);
 		}
 
+		// Get the message node which was added
```

Comment:
This methodHook needs to be run after the original method otherwise this will not work (or at least based on the addChatMessage code, it is added to buffer from inside of it and the method also assumes that buffer can be null and creates it when needed)